core/src/main/java/com/sopt/core/designsystem/component/calendar/NoostakCalendar.kt

```diff
@@ -0,0 +1,289 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.sopt.core.R
+import com.sopt.core.designsystem.component.snackbar.NoostakSnackBar
+import com.sopt.core.designsystem.theme.NoostakTheme
+import com.sopt.core.extension.noRippleClickable
+import java.time.LocalDate
+import java.time.YearMonth
+
+@Composable
+fun NoostakCalendar(
+    start: String,
+    end: String,
+    isSingleDate: Boolean,
+    // isRangeSelected: (String, String) -> Unit,
+    isRangeSelected: (List<String>) -> Unit,
+    modifier: Modifier = Modifier,
+    days: List<String>
+) {
+    val typography = NoostakTheme.typography
+    val colors = NoostakTheme.colors
+
+    var year by remember { mutableStateOf(LocalDate.now().year) }
+    var month by remember { mutableStateOf(LocalDate.now().monthValue) }
+    var selectedDates by remember { mutableStateOf<List<String>>(emptyList()) }
+    var startDate by remember { mutableStateOf(start) }
+    var endDate by remember { mutableStateOf(end) }
+    var showMessage by remember { mutableStateOf(false) }
+
+    val yearMonth = YearMonth.of(year, month)
+    val totalDays = yearMonth.lengthOfMonth()
+    val firstDay = LocalDate.of(year, month, 1).dayOfWeek.value % 7
+
+    LaunchedEffect(isSingleDate) {
+        selectedDates = emptyList()
+        startDate = ""
+        endDate = ""
+    }
+
+    LaunchedEffect(selectedDates, startDate, endDate) {
+        if (isSingleDate) {
+            isRangeSelected(selectedDates)
+        } else if (startDate.isNotEmpty() && endDate.isNotEmpty()) {
+            val startLocalDate = LocalDate.parse(startDate)
+            val endLocalDate = LocalDate.parse(endDate)
+            val rangeDates = (0..endLocalDate.toEpochDay() - startLocalDate.toEpochDay())
+                .map { startLocalDate.plusDays(it).toString() }
+            isRangeSelected(rangeDates)
+        }
+    }
+
+    LaunchedEffect(showMessage) {
+        if (showMessage) {
+            kotlinx.coroutines.delay(3000)
+            showMessage = false
+        }
+    }
+
+    Column(modifier = modifier) {
+        Row(
+            modifier = Modifier.align(Alignment.CenterHorizontally),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Image(
+                painter = painterResource(id = R.drawable.ic_calendar_left),
```

Comment:
P1: 좌우 아이콘 imagevector로 수정해주세요
```suggestion
                imageVector = ImageVector.vectorResource(id = R.drawable.ic_calendar_left),
```